src/vivarium_csu_zenon/components/treatment/treatment.py

```diff
@@ -209,3 +208,71 @@ def load_adherence_p(builder):
+        return 'ldlc_treatment_effect'
+
+    def setup(self, builder: 'Builder'):
+        self.treatment_effect = self.load_treatment_effect(builder)
+
+        self.is_adherent = builder.value.get_value('ldlc_treatment_adherence')
+        self.columns_required = [parameters.STATIN_LOW, parameters.STATIN_HIGH,
+                                 parameters.FIBRATES, parameters.EZETIMIBE, parameters.LIFESTYLE]
+
+        # This pipeline is not required.  It's a convenience for reporting later.
+        self.effect_size = builder.value.register_value_producer(self.name, self.compute_effect_size,
+                                                                 requires_columns=self.columns_required,
+                                                                 requires_values=['ldlc_treatment_adherence'])
+
+        builder.value.register_value_modifier('high_ldl_cholesterol.exposure',
+                                              self.adjust_exposure,
+                                              requires_values=[self.name])
+        self.population_view = builder.population.get_view(self.columns_required + ['initial_treatment_effect_size'])
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 creates_columns=['initial_treatment_effect_size'],
+                                                 requires_values=[self.name])
+
+    def on_initialize_simulants(self, pop_data: 'SimulantData'):
+        self.population_view.update(pd.Series(self.effect_size(pop_data.index),
+                                              index=pop_data.index,
+                                              name='initial_treatment_effect_size'))
+
+    def adjust_exposure(self, index, exposure):
+        initial_effect = self.population_view.subview(['initial_treatment_effect_size']).get(index)
+        initial_effect = initial_effect['initial_treatment_effect_size']  # coerce df to series
+        return exposure + initial_effect - self.effect_size(index)
+
+    def compute_effect_size(self, index: pd.Index):
+        pop_status = self.population_view.subview(self.columns_required).get(index)
+        effect_size = pd.Series(0, index=index)
+
+        # potency_statin_dose
+        treatment_profiles = {
+            parameters.HIGH_STATIN_HIGH: pop_status[parameters.STATIN_HIGH] == 'high',
+            parameters.HIGH_STATIN_LOW: pop_status[parameters.STATIN_HIGH] == 'low',
+            parameters.LOW_STATIN_HIGH: pop_status[parameters.STATIN_LOW] == 'high',
+            parameters.LOW_STATIN_LOW: pop_status[parameters.STATIN_LOW] == 'low',
+            parameters.EZETIMIBE: pop_status[parameters.EZETIMIBE],
+            parameters.FIBRATES: pop_status[parameters.FIBRATES],
+            parameters.LIFESTYLE: pop_status[parameters.LIFESTYLE],
+        }
+
+        for treatment, mask in treatment_profiles.items():
+            effect_size.loc[mask] += self.treatment_effect[treatment]
```

Comment:
see comment above about LLT effect being multiplicative; based on that I was expecting something like `effect_size.loc[mask] *= (1-self.treatment_effect[treatment])`